src/Bishop.cpp

```diff
@@ -0,0 +1,74 @@
+#include "Bishop.h"
+
+Bishop::Bishop(int color, Board* board, int row, int col) :Piece(color, board, row, col) {}
+
+bool Bishop::isValidMove(int target_row, int target_col) {
+	if (_row == target_row || _col == target_col)
+		return false;
+	return Piece::isValidMove(target_row, target_col);
+}
+
+
+std::vector<std::tuple<int, int>> Bishop::getAllvalidMoves() {
+	std::vector<std::tuple<int, int>> moves;
+	traverseleftToRightDiagonal(moves);
+	traverseRightToLeftDiagonal(moves);
+	return moves;
+}
+
+void Bishop::traverseleftToRightDiagonal(std::vector<std::tuple<int, int>>& moves) {
+	for (int i = _row + 1, j = _col + 1; i < ROW_LENGTH && j < COL_LENGTH; i++, j++)
+	{
```

Comment:
bad way for checking collisions, pieces have full access for the board

Reply:
I do not understand the comment.